platform/email/platform-email-impl/src/test/java/org/codice/ddf/platform/email/impl/EmailServiceImplTest.java

```diff
@@ -0,0 +1,188 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.email.impl;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.ServerSocket;
+import java.util.Iterator;
+
+import javax.activation.DataHandler;
+import javax.activation.FileDataSource;
+import javax.mail.BodyPart;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+
+import org.junit.Test;
+
+import com.dumbster.smtp.SimpleSmtpServer;
+import com.dumbster.smtp.SmtpMessage;
+
+public class EmailServiceImplTest {
+
+    private static final String HOSTNAME = "127.0.0.1";
+
+    private static final String ATTACHMENT_TEXT = "text text text";
+
+    private static final String ATTACHMENT_FILENAME = "myFile.txt";
+
+    private static final String SUBJECT = "sample subject";
+
+    private static final String TO_ADDR = "user@example.com";
+
+    private static final String FROM_ADDR = "nobody@example.com";
+
+    private static final String BODY = "test body";
+
+    private static final String SUBJECT_HEADER = "Subject";
+
+    private static final String FROM_HEADER = "From";
+
+    private static final String TO_HEADER = "To";
+
+    @Test
+    public void testSendWithAttachments() throws IOException, MessagingException {
+
+        int port = findAvailablePort();
```

Comment:
I always get concerned when I see ports being opened in unit tests.  Can we use a mock `Session` instead to avoid needing to startup a SMTP server?